src/lib/work_selector/intf.ml

```diff
@@ -153,7 +153,7 @@ module type Selection_method_intf = sig
 
   val pending_work_statements :
        snark_pool:snark_pool
-    -> fee:Currency.Fee.t
+    -> fee_opt:Currency.Fee.t option
```

Comment:
An alternative consideration would be to change this type signature to `?fee:Currency.Fee.t -> snark_pool:snark_pool -> staged_ledger:staged_ledger -> unit -> Transaction_snark.Statement.t One_or_two.t list`.

Reply:
We want to be able to differentiate between a default fee and no fee for when the node is not doing any snark work